core/domain/question_services.py

```diff
@@ -56,8 +58,12 @@ def _migrate_state_schema(versioned_question_state):
             feconf.CURRENT_STATE_SCHEMA_VERSION)
 
     while state_schema_version < feconf.CURRENT_STATE_SCHEMA_VERSION:
-        question_domain.Question.update_state_from_model(
-            versioned_question_state, state_schema_version)
+        if state_schema_version == 29:
```

Comment:
Try not to branch here. The comment I left above (re not having to pass in question_model) should make it simpler.

Reply:
Removed.